### PR TITLE
TW-133: swap section percentage

### PR DESCRIPTION
**Pull request by letier3110**

### CODE REVIEW

File: src/navigator/navigation-bar/side-bar/side-bar.tsx [hunk 1]
```diff
@@ -79,6 +98,16 @@ export const SideBar: FC<Props> = ({ currentRouteName }) => {
...
         </View>
+        <KeyboardAvoidingView
+          keyboardVerticalOffset={isAndroid ? formatSize(60) : formatSize(50)}
+          behavior="padding"
+          style={{
+            height: formatSize(44)
+          }}
+        >
+          {isKeyboardVisible && <View style={styles.keyboardContainer} />}
+          {/* <View style={styles.keyboardContainer} /> */}
```
Comment (unixvb) [inline src/navigator/navigation-bar/side-bar/side-bar.tsx:109]: delete commented code

File: src/screens/swap/swap-form/swap-percentage/swap-percentage.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import { getTokenSlug } from '../../../../token/utils/token.utils';
+import { useSwapPercentageStyles } from './swap-percentage.styles';
+
+interface Props {
+  formik: FormikState<SwapFormValues> & {
+    setFieldValue: (field: string, value: unknown, shouldValidate?: boolean) => void;
+  };
+}
+
+export const SwapPercentage: FC<Props> = ({ formik }) => {
```
Comment (unixvb) [inline src/screens/swap/swap-form/swap-percentage/swap-percentage.tsx:20]: this also should be available on Send screen.  
Current implementation is not reusable.

File: src/components/percentage-selector/percentage-selector.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+      keyboardDidShowListener.remove();
+    };
+  }, []);
+
+  return (
+    <KeyboardAvoidingView
+      keyboardVerticalOffset={isKeyboardVisible ? (isAndroid ? formatSize(80) : formatSize(37.5)) : 0}
+      behavior="padding"
+      // eslint-disable-next-line react-native/no-inline-styles
+      style={[percentageStyles.keyboard, { height: isKeyboardVisible ? formatSize(44) : 0 }]}
```
Comment (LendiHop) [inline src/components/percentage-selector/percentage-selector.tsx:40]: 1. style={[percentageStyles.keyboard, isKeyboardVisible && { height: formatSize(44) }]}
2. remove eslint rule

File: src/components/percentage-selector/percentage-selector.styles.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { createUseStyles } from '../../styles/create-use-styles';
+import { formatSize } from '../../styles/format-size';
+
+export const usePercentageSelectorStyles = createUseStyles(({ colors, typography }) => ({
+  keyboard: {
+    flexDirection: 'row',
+    alignItems: 'center',
+    height: formatSize(44)
```
Comment (LendiHop) [inline src/components/percentage-selector/percentage-selector.styles.ts:8]: height: 0

File: src/components/percentage-selector/percentage-selector.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+    return () => {
+      keyboardDidHideListener.remove();
+      keyboardDidShowListener.remove();
+    };
+  }, []);
+
+  return (
+    <KeyboardAvoidingView
+      keyboardVerticalOffset={isKeyboardVisible ? (isAndroid ? formatSize(80) : formatSize(37.5)) : 0}
```
Comment (LendiHop) [inline src/components/percentage-selector/percentage-selector.tsx:37]: const androidKeyboardVerticalOffset = isAndroid ? formatSize(80) : formatSize(37.5); + useMemo. And reuse it here

File: src/components/percentage-selector/percentage-selector.tsx [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+                handleChange(newValue);
+                Keyboard.dismiss();
+              }}
+              style={percentageStyles.percentageShape}
+            >
+              <Text style={percentageStyles.percentageText}>75%</Text>
+            </TouchableOpacity>
+            <TouchableOpacity
+              onPress={() => {
+                const newValue = BigNumber.maximum(new BigNumber(balance).minus(symbol === 'TEZ' ? 300000 : 0), 0);
```
Comment (LendiHop) [inline src/components/percentage-selector/percentage-selector.tsx:77]: useMemo?

File: src/components/percentage-selector/percentage-selector.tsx [hunk 7]
```diff
@@ -32,62 +32,60 @@ export const PercentageSelector: FC<Props> = ({ symbol, balance, handleChange })
...
+
+  const handle75Percentage = useCallback(() => handlePercentage(0.75), [handlePercentage]);
+
+  const handleMax = useCallback(() => {
+    const newValue = BigNumber.maximum(new BigNumber(balance).minus(symbol === 'TEZ' ? 300000 : 0), 0);
+    handleChange(newValue);
+    Keyboard.dismiss();
+  }, [handleChange, balance, symbol]);
+
+  const handleDismiss = useCallback(() => {
```
Comment (LendiHop) [inline src/components/percentage-selector/percentage-selector.tsx:62]: short return

**Review by LendiHop — APPROVED** [review]

**Review by LendiHop — APPROVED** [review]

File: src/components/percentage-selector/percentage-selector.tsx [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    (percentage: number) => () => {
+      const newValue = new BigNumber(balance).times(percentage);
+      handleChange(newValue);
+      Keyboard.dismiss();
+    },
+    [handleChange, balance]
+  );
+
+  const handleMax = useCallback(() => {
+    const newValue = BigNumber.maximum(new BigNumber(balance).minus(symbol === 'TEZ' ? 300000 : 0), 0);
```
Comment (unixvb) [inline src/components/percentage-selector/percentage-selector.tsx:51]: create a variable that will explain what `condition ? 300000 : 0` value means

**Review by LendiHop — APPROVED** [review]

**Review by unixvb — APPROVED** [review]

**Pull request closed without merging**